### ISSUE TITLE
Code Editor 2: when using specific amount of diacritics in CE2, code highlight is broken

### ISSUE THREAD

**Issue by gnysek** [YoYoGames/GameMaker-Bugs#8112]
### Description
In some cases, when using non-latin characters, highlighting strings breaks. It's either caused by specific characters, or by specific amount of them.

For example, this code shows strings in white, and some part of string is red:
```
global._LANG = ds_map_create();
function lang_define_TEMP() {
	global._LANG[? "INFO"] = "!!! INFO !!!\nZażółć gęślą.\nRegulartext.\n\nLolo";
	global._LANG[? "yes"] = "Yes";
	global._LANG[? "No"] = "No";
}
```

![Image](https://github.com/user-attachments/assets/8bc83d2a-bc0f-4082-a8b5-b2f02824dc51)

### Steps To Reproduce
1. Put above code in any script (same code is attached as project)
2. Try replacing `ą` with `a` (in word `gęślą`, line 3). Even when there are other diacritics characters in text, it now looks ok.

### Which version of GameMaker are you reporting this issue for?
IDE v2024.1100.0.634 Runtime v2024.1100.0.658

### Which operating system(s) are you seeing the problem on?
Windows 10.0.22631.0

[63a7a29d-0454-44b7-b448-56136f1b7fe7](https://api.gamemaker.io/api/github/downloads/63a7a29d-0454-44b7-b448-56136f1b7fe7)

**Comment by gnysek** [comment]
Might be same issue as in #8111 .

**Comment by zreedy** [comment]
Related https://github.com/YoYoGames/GameMaker-Bugs/issues/8097 and https://github.com/YoYoGames/GameMaker-Bugs/issues/8111 and https://github.com/YoYoGames/GameMaker-Bugs/issues/8097 and https://github.com/YoYoGames/GameMaker-Bugs/issues/8112

**Comment by zreedy** [comment]
Fixed. These tickets share the same underlying issue: https://github.com/YoYoGames/GameMaker-Bugs/issues/6456 https://github.com/YoYoGames/GameMaker-Bugs/issues/8112 https://github.com/YoYoGames/GameMaker-Bugs/issues/8111

The char offset was converted to a byte offset in the UTF-8 when passed to the Regex library, but was not converted from byte offset to char offset when returned to the TextMate Grammar Tokenizer which caused the tokenizer to incorrectly parse the string by skipping characters.

**Comment by zreedy** [comment]
Additional changes for this ticket include adding a IDE Fallback Font which is under "General Settings" and allows users to specify what font to use when the glyph is not found in the primary font (IDE Font for labels, CE2 Theme font for text in code editor).

By default this is set to "Open Sans" which we distribute on all platforms. Chinese users are best to set this to "Microsoft YaHei" on Windows (environment default), Japanese users will find that "Meiryo" is best (environment default). Please remember that the kanji/hanzi codepage overlaps and the correct rendering of these glyphs depends on the font selected.

**Comment by zreedy** [comment]
Fixed + Merged

**Comment by Emc1923** [comment]
Verified fixed in  IDE v2024.1300.0.714  Runtime v2024.1300.0.727
